mlflow/models/utils.py

```diff
@@ -906,7 +920,15 @@ def _is_scalar(x):
                     )
         elif isinstance(pf_input, (list, np.ndarray, pd.Series)):
             pf_input = pd.DataFrame(pf_input)
-
+        elif isinstance(pf_input, SparkDataFrame):
+            for column, dtype in pf_input.dtypes:
+                if "array" in dtype or "map" in dtype:
+                    _logger.warning(
+                        "The input data contains array or map type. Note that"
+                        " array and map type are not supported in schema validation"
+                        " when using pyspark DataFrame as input"
+                    )
```

Comment:
Why not support ArrayType? MLflow model signatures now support arrays of primitives, arrays of arrays, and arrays of structs. cc @serena-ruan 

We will also add MapType support soon

Reply:
There are issues with converting between spark and pandas for specific array and map types. See [this doc.](https://docs.databricks.com/en/pandas/pyspark-pandas-conversion.html#supported-sql-types) Additionally for feature store models we do not add arrayTypes and mapTypes to the signature so there wouldn't be any schema validation happening regardless (for the feature store case).

Reply:
Choosing not to support these types because currently feature store models do not log these types into their signature. Enforcing schema for these types would cause incompatibility with previously logged feature store models. May opt to skip schema enforcement if these types are detected.

Reply:
added support for these types